service/lib/agama/software/manager.rb

```diff
@@ -597,8 +598,19 @@ def missing_registration_issue
       #
       # @return [Boolean]
       def missing_registration?
+        return false unless product
+
         registration.reg_code.nil? &&
-          registration.requirement == Agama::Registration::Requirement::MANDATORY
+          product.registration &&
+          missing_base_product?
```

Comment:
just to be sure, do we update issue when product is change or when repository is added?

Reply:
Every time we reevaluate the software layer (which includes registration, repositories probing, etc.). But it might happen what we miss some case, we need to do more testing.